app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -1,4 +1,6 @@
 class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  skip_before_action :verify_authenticity_token, :only => [:failure]
```

Comment:
Why are skipping the `verify_authenticity_token`?

Reply:
mastodon-omniauth [does not include a CSRF token](https://github.com/tootsuite/omniauth-mastodon/blob/master/lib/omniauth/strategies/mastodon.rb#L63-L66), because [omniauth-oauth2 refuses to handle the request if a csrf is present](https://github.com/omniauth/omniauth-oauth2/blob/master/lib/omniauth/strategies/oauth2.rb#L71).
But Rails refuses to handle the failure if there's no token…